### ISSUE TITLE
[BUG] subtitle is not displayed 

### ISSUE THREAD

**Issue by avinh** [jhomlala/betterplayer#296]
Subtitles are not displayed if the subtitle line has 2 or more lines

**Comment by jhomlala** [comment]
@avinh please fill issue template properly. You didn't provide all necessary information. Also please provide subtitles that you're using.

**Comment by avinh** [comment]
Image in better player
https://i.ibb.co/103M9Bx/photo-2021-02-13-05-32-13.jpg
Image in JWPlayer web browser 
https://i.ibb.co/d2TZjFv/image.png

I mean, if the caption is too long, it will show up on 2 lines, in better player it doesn't display

**Comment by jhomlala** [comment]
@avinh Please provide link to video or subtitles.

**Comment by avinh** [comment]
Yes, this is HLS video. You can use any video, or any subtitle
https://bitdash-a.akamaihd.net/content/sintel/hls/playlist.m3u8
https://bitdash-a.akamaihd.net/content/sintel/hls/subtitles_en.vtt

1 line 
`
2
00:01:51.800 --> 00:01:55.800
It has shed much innocent blood.
`
2 line (not displayed)
`
3
00:01:58.000 --> 00:02:01.450
You're a fool for traveling alone,
so completely unprepared.
`

**Comment by jhomlala** [comment]
Subtitles are working fine. I've tested both subtitles from HLS and file.
![image](https://user-images.githubusercontent.com/1329033/107970365-502db780-6fb1-11eb-8e68-9273cb4df4ad.png)
![image](https://user-images.githubusercontent.com/1329033/107970355-4dcb5d80-6fb1-11eb-90cc-24eb56d5caba.png)

To use HLS subtitles you need to choose them from overflow menu
To use VTT subtitles you need to pass subtitles data source:

```
 subtitles: BetterPlayerSubtitlesSource.single(
          url:
              "https://bitdash-a.akamaihd.net/content/sintel/hls/subtitles_en.vtt",
          selectedByDefault: true,
          type: BetterPlayerSubtitlesSourceType.network),
    );
```
From my perspective there's no issue.

**Comment by jhomlala** [comment]
Feel free to answer in this issue, so I will reopen it.

**Comment by avinh** [comment]
I checked the subtitle above and it didn't error, so sorry about that
I got an error in this subtitle, please check it with the movie: The Croods: A New Age
http://soociu.top/subtitles/moviexWXewJnnbrbPsNSyrAnl389.vtt

**Comment by avinh** [comment]
It appears log like this for error lines
![](https://i.ibb.co/dmjPHVY/image.png)

**Comment by m1n093** [comment]
![image](https://user-images.githubusercontent.com/41908373/108134643-d54ec480-70b6-11eb-9e18-1137225a19e7.png)

```
static BetterPlayerSubtitle _handle3LinesAndMoreSubtitles(
      List<String> scanner) {
    try {
      if (scanner[0].isEmpty) {
        scanner.removeAt(0);
      }

      final index = int.tryParse(scanner[0]);

      final timeSplit = scanner[1].split(timerSeparator);
      final start = _stringToDuration(timeSplit[0]);
      final end = _stringToDuration(timeSplit[1]);
      final texts = scanner.sublist(2, scanner.length);

      return BetterPlayerSubtitle._(
          index: index, start: start, end: end, texts: texts);
    } catch (exception) {
      BetterPlayerUtils.log("Failed to parse subtitle line: $scanner");
      return BetterPlayerSubtitle._();
    }
  }
```

Why do you expect line 0 is empty? In my vtt subtitles I have line 0 with duration and it cause exception with "Failed to parse subtitle line: "

**Comment by jhomlala** [comment]
Fixed issue in 0.0.56.
